src/impy/models/dpmjetIII.py

```diff
@@ -136,17 +125,26 @@ def _cross_section(self, kin=None, precision=None):
             if precision is not None:
                 self._lib.dtglgp.jstatb = saved
             return CrossSectionData(inelastic=self._lib.dtglxs.xspro[0, 0, 0])
-        else:
-            stot, sela = self._lib.dt_xshn(
-                self._lib.idt_icihad(kin.p1), self._lib.idt_icihad(kin.p2), 0.0, kin.ecm
-            )
-            return CrossSectionData(total=stot, elastic=sela, inelastic=stot - sela)
+
+        assert kin.p1.is_hadron
+        assert kin.p2.is_hadron
+        target_id = self._lib.idt_icihad(kin.p2)
+        stot, sela = self._lib.dt_xshn(projectile_id, target_id, 0.0, kin.ecm)
+        return CrossSectionData(total=stot, elastic=sela, inelastic=stot - sela)
 
         # TODO set more cross-sections
 
     def _set_kinematics(self, kin):
         # Save maximal mass that has been initialized
         # (DPMJET sometimes crashes if higher mass requested than initialized)
+
+        # Relax momentum and energy conservation checks at very high energies
+        if kin.ecm > 5e4:
```

Comment:
Can't do that without setting it back. Like this the output will depend on the sequence of requested energies since energy-momentum conservation checks can reject. Either this has to stay in init or, one needs to save the default values and reset them if kinematics change.

Reply:
It cannot be in init. 

Ah, I see what you mean, but it does not really matter, I think. We need to use MCRunRemote which always uses a fresh copy of dpmjet when _set_kinematics is called, so it is always called only once per dpmjet instance.

Reply:
I tried to fix this a bit, but you are the DPMJet expert, so I am relying on your help.